ansible/templates/cluster-template.yaml.j2

```diff
@@ -532,75 +386,30 @@ worker:
-#      awsEnvironment:
-#        enabled: true
-#        environment:
-#          CFNSTACK: '{ "Ref" : "AWS::StackId" }'
-#
 #      # Add predefined set of labels to the nodes
 #      # The set includes names of launch configurations and autoscaling groups
       awsNodeLabels:
         enabled: true
-#
 #      # Will provision worker nodes with IAM permissions to run cluster-autoscaler and add node labels so that
 #      # cluster-autoscaler pods are scheduled to nodes in this node pool
       clusterAutoscalerSupport:
         enabled: true
-#
-#      # Mount an EFS only to a specific node pool.
-#      # This is a NFS share that will be available across a node pool through a hostPath volume on the "/efs" mountpoint
-#      #
-#      # If you'd like to configure an EFS mounted to every node, use the top-level `elasticFileSystemId` instead
-#      #
-#      # Beware that currently it doesn't work for a node pool in subnets managed by kube-aws.
-#      # In other words, this node pool must be in existing subnets managed by you by specifying subnets[].id to make this work.
-#      elasticFileSystemId: fs-47a2c22e
-#
-#      # This option has not yet been tested with rkt as container runtime
-#      ephemeralImageStorage:
-#        enabled: true
-#
-#      # When enabled it will grant sts:assumeRole permission to the IAM roles for worker nodes.
-#      # This is intended to be used in combination with managedIamRoleName. See #297 for more information.
-#      kube2IamSupport:
-#        enabled: true
-#
-#      # Propagate custom CLI options to kubelet
-#      # Provided example overrides default docker image garbage collection limits
-#      # as in https://kubernetes.io/docs/concepts/cluster-administration/kubelet-garbage-collection/
-#      # Anything from https://kubernetes.io/docs/admin/kubelet/ can be used as a value
-#
       kubeletOpts: --image-gc-low-threshold=65 --image-gc-high-threshold=75
-#
+
 #      # Kubernetes node labels to be added to worker nodes
       nodeLabels:
-        kube-aws.coreos.com/role: worker
-#
-#      # Kubernetes node taints to be added to worker nodes
-#      taints:
-#        - key: dedicated
-#          value: search
-#          effect: NoSchedule
-#
-#      # Other less common customizations per node pool
-#      # All these settings default to the top-level ones
-#      keyName:
-#      releaseChannel: alpha
-#      amiId:
-#      kubernetesVersion: 1.6.0-alpha.1
-#
-#      # Images are taken from controlplane by default, but you can override values for node pools here. E.g.:
-#      AwsCliImage:
-#        repo: quay.io/coreos/awscli
-#        tag: edge
-#        rktPullDocker: false
-#
-#      sshAuthorizedKeys:
-#      # User-provided YAML map available in control-plane's stack-template.json
-#      customSettings:
-#        key1: [ 1, 2, 3 ]
+        kube-aws.coreos.com/role: {{ worker.role }}
+{% if 'mongo-worker' == worker.role %}
```

Comment:
I'm not a big fan of this if. Can we instead introduce a new property in the dict like dedicated-taints and do the if on it?